internal/server/upload.go

```diff
@@ -0,0 +1,152 @@
+package server
+
+import (
+	"encoding/csv"
+	"errors"
+	"fmt"
+	"github.com/opg-sirius-finance-admin/internal/api"
+	"github.com/opg-sirius-finance-admin/internal/model"
+	"github.com/opg-sirius-finance-admin/internal/util/util"
+	"io"
+	"net/http"
+	"os"
+	"reflect"
+	"strings"
+	"unicode"
+)
+
+type UploadHandler struct {
+	router
+}
+
+func (h *UploadHandler) render(v AppVars, w http.ResponseWriter, r *http.Request) error {
+	ctx := getContext(r)
+
+	//Create and defer cleanup of a temp directory
+	dir := "./uploads"
+	if err := os.MkdirAll(dir, os.ModePerm); err != nil {
+		http.Error(w, "Failed to create directory", http.StatusInternalServerError)
+		return err
+	}
+	defer func() {
+		if err := os.RemoveAll(dir); err != nil {
+			fmt.Println("Could not remove directory:", err)
+		}
+	}()
+
+	// Extract form data
+	reportUploadType := r.PostFormValue("reportUploadType")
+	uploadDate := r.PostFormValue("uploadDate")
+	email := r.PostFormValue("email")
+
+	// Handle file upload
+	file, headers, err := r.FormFile("fileUpload")
+	if err != nil {
+		return h.handleError(w, r, "No file uploaded", http.StatusBadRequest)
+	}
+	defer file.Close()
+
+	// Define the destination file path
+	dst, err := os.Create(headers.Filename)
```

Comment:
do you need to create a real file here, or would you be able to get away with using the data?